### PR TITLE
Expose esx hosts

### PR DESCRIPTION
**Pull request by mmidolesov**

### CODE REVIEW

File: nova/compute/manager.py [hunk 1]
```diff
@@ -7252,7 +7252,7 @@ def _get_nodename(self, instance, refresh=False):
         on this host. This method should not be used with any operations
         on ironic instances since it does not handle multiple nodes.
         """
-        node = self.driver.get_available_nodes(refresh=refresh)[0]
```
Comment (fwiesel) [inline nova/compute/manager.py]: Could you keep that interface?
It would affect other hypervisors too.

File: nova/compute/manager.py [hunk 2]
```diff
@@ -1822,7 +1822,6 @@ def _do_build_and_run_instance(self, context, instance, image,
             request_spec, filter_properties, admin_password, injected_files,
             requested_networks, security_groups, block_device_mapping,
             node=None, limits=None, host_list=None):
-
```
Comment (fwiesel) [inline nova/compute/manager.py]: Could you remove also the white-space changes, like this one?

Comment (mmidolesov) [reply nova/compute/manager.py]: Sure sorry about that.

**Comment by mmidolesov** [comment]
Closing pull request since new one was created with resolved conflicts and containing current changes with the additional ones in it.

**Pull request closed without merging**